File: core/src/remove/tanks/game/level/utility/create/CreateEntryPrefab.java
package remove.tanks.game.level.utility.create;

import com.google.common.base.MoreObjects;
import remove.tanks.game.utility.Prefab;
import remove.tanks.game.utility.surface.position.PositionPrefab;

/**
 * @author Mateusz Długosz
 */
public final class CreateEntryPrefab extends Prefab {
    private final String prefabCode;
    private final PositionPrefab positionPrefab;

    public CreateEntryPrefab(String prefabCode, PositionPrefab positionPrefab) {
        this.prefabCode = prefabCode;
        this.positionPrefab = positionPrefab;
    }

    public String getPrefabCode() {
        return prefabCode;
    }

    public PositionPrefab getPositionPrefab() {
        return positionPrefab;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("prefabCode", prefabCode)
                .add("positionPrefab", positionPrefab)
                .toString();
    }
}

File: tools/docs-generator/lib/level/utility/create/create_entry_prefab.py
class CreateEntryPrefab(object):
    def __init__(self, prefab_code, position_prefab):
        self.prefab_code = str(prefab_code)
        self.position_prefab = position_prefab

    def get_prefab_code(self):
        return self.prefab_code

    def get_position_prefab(self):
        return self.position_prefab

    def __str__(self):
        return "CreateEntryPrefab(prefab_code={}, position_prefab={})" \
            .format(self.prefab_code, str(self.position_prefab))
